src/lib/select/select.ts

```diff
@@ -242,6 +245,15 @@ export class MdSelect implements AfterContentInit, ControlValueAccessor, OnDestr
   ngAfterContentInit() {
     this._initKeyManager();
     this._resetOptions();
+
+    // Assign any values that were deferred until the component is initialized.
+    if (this._tempValue) {
+      Promise.resolve(null).then(() => {
+        this.writeValue(this._tempValue);
+        this._tempValue = null;
+      });
+    }
+
     this._changeSubscription = this.options.changes.subscribe(() => {
```

Comment:
Makes sense, I'll rework it. This was already being handled in a similar way on the multiple selection PR.